eventstore/eventstore.go

```diff
@@ -39,32 +46,129 @@ type EventStore struct {
 // Open opens the event store. It should be closed later with the
 // Close() method.
 func Open(fileName string) (*EventStore, error) {
-	db, err := bolt.Open(fileName, 0600, nil)
+	db, err := bolt.Open(fileName, 0600, &bolt.Options{Timeout: openTimeout})
 	if err != nil {
 		return nil, err
 	}
 
+	log.Println("getting events to migrate from old bucket")
+	eventsToMigrate, oldEventTimes, err := getEventsToMigate(db)
+	if err != nil {
+		return nil, err
+	}
+	log.Printf("got %d events to migrate from old bucket\n", len(eventsToMigrate))
+
 	err = db.Update(func(tx *bolt.Tx) error {
-		_, err := tx.CreateBucketIfNotExists(bucketName)
+		_, err := tx.CreateBucketIfNotExists(oldBucketName)
+		if err != nil {
+			return err
+		}
+		_, err = tx.CreateBucketIfNotExists(idDataBucketName)
 		return err
 	})
 	if err != nil {
 		db.Close()
 		return nil, fmt.Errorf("creating bucket: %v", err)
 	}
 
-	return &EventStore{
-		db: db,
-	}, nil
+	store := &EventStore{db: db}
+	for _, event := range eventsToMigrate {
+		// Adding/Migrating to new bucket
+		if err := store.Add(&event); err != nil {
+			return nil, err
+		}
+	}
+	log.Println("migrated all old events")
+
+	// Delete events from old bucket after migrate
+	err = db.Update(func(tx *bolt.Tx) error {
+		oldBucket := tx.Bucket(oldBucketName)
+		if oldBucket == nil {
+			return nil
+		}
+		for _, oldEventTime := range oldEventTimes {
+			log.Printf("deleting %v", oldEventTime.Details)
+			if err := oldBucket.Delete(oldEventTime.Details); err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Strongly recommend you move this to a private migrate method. At the very least it'll make all the error returns cleaner (just one thing to return).